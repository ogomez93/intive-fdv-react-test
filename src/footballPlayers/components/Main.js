import React from 'react';
import Filters from './Filters';
import Table from './Table';

const Main = () => (
  <div>
    <h2 className="mb-2">Football Player Finder</h2>
    <Filters />
    <Table />
  </div>
);

export default Main;