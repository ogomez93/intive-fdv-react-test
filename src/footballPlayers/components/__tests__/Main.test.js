import React from 'react';
import { shallow } from 'enzyme';

import Main from '../Main';

jest.mock('../Filters', () => 'Filters');
jest.mock('../Table', () => 'Table');

describe('Component: <Main />', () => {
  let wrapper, title, filters, table;

  beforeAll(() => {
    wrapper = shallow(<Main />);
    title = wrapper.find('h2');
    filters = wrapper.find('Filters');
    table = wrapper.find('Table');
  });

  it('should render', () =>
    expect(wrapper.isEmptyRender()).toEqual(false));

  it('should render correctly', () =>
    expect(wrapper).toMatchSnapshot());

  it('should render the title', () =>
    expect(title.text()).toEqual('Football Player Finder'));

  it('should render the Filters component', () => 
    expect(filters).not.toBeNull());

  it('should render the Table component', () => 
    expect(table).not.toBeNull());
});
