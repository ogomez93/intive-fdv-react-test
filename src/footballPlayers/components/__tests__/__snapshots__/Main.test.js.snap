// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Main /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="mb-2"
>
          Football Player Finder
</h2>,
        <Filters />,
        <Table />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Football Player Finder",
          "className": "mb-2",
        },
        "ref": null,
        "rendered": "Football Player Finder",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "Filters",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "Table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="mb-2"
>
            Football Player Finder
</h2>,
          <Filters />,
          <Table />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Football Player Finder",
            "className": "mb-2",
          },
          "ref": null,
          "rendered": "Football Player Finder",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "Filters",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "Table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
